src/BloomExe/Edit/EditingModel.cs

```diff
@@ -1196,26 +1188,28 @@ private string GetPathToCurrentTemplateHtml
 			}
 		}
 
-		private const string URL_PREFIX = "/bloom/localhost/";
-		private const string JSON_START = "[{ ";
-		private const string JSON_DIVIDER = " , ";
-		private const string JSON_END = " }]";
-
 		private string GetJsonTemplatePageObject
 		{
 			get
 			{
-				var folderPath = Path.GetDirectoryName(GetPathToCurrentTemplateHtml);
-				folderPath = URL_PREFIX + folderPath.Replace(':', '$').Replace('\\', '/');
-				var jsonBookFolder = "\"templateBookFolderUrl\": \"" + folderPath + "\"";
-				var htmlFilePath = URL_PREFIX + GetPathToCurrentTemplateHtml;
-				htmlFilePath = htmlFilePath.Replace(':', '$').Replace('\\', '/');
-				var jsonBook = "\"templateBookUrl\": \"" + htmlFilePath + "\"";
-				var jsonLastPage = "\"lastPageAdded\": \"" + _lastPageAdded + "\"";
-				var jsonString = JSON_START + jsonBookFolder + JSON_DIVIDER + jsonBook + JSON_DIVIDER + jsonLastPage + JSON_END;
-				return jsonString;
+				var addPageSettings = new Dictionary<string, object>();
+				var folderPath = MassageUrlForJavascript(Path.GetDirectoryName(GetPathToCurrentTemplateHtml));
+				addPageSettings.Add("templateBookFolderUrl", folderPath);
+				var htmlFilePath = MassageUrlForJavascript(GetPathToCurrentTemplateHtml);
+				addPageSettings.Add("templateBookUrl", htmlFilePath);
+				addPageSettings.Add("lastPageAdded", _lastPageAdded);
+				var settingsString = JsonConvert.SerializeObject(new object[] {addPageSettings});
+				return settingsString;
```

Comment:
As Add Page is currently set up it requires an array of collectionUrl objects. I can't figure out how to get the JsonConvert.SerializeObject method to generate an array other than the way I did it here.